dynamic_programming/subset_generation.py

```diff
@@ -1,44 +1,25 @@
-# Print all subset combinations of n element in given set of r element.
-
-
-def combination_util(arr, n, r, index, data, i):
-    """
-    Current combination is ready to be printed, print it
-    arr[]  ---> Input Array
-    data[] ---> Temporary array to store current combination
-    start & end ---> Staring and Ending indexes in arr[]
-    index  ---> Current index in data[]
-    r ---> Size of a combination to be printed
-    """
-    if index == r:
-        for j in range(r):
-            print(data[j], end=" ")
-        print(" ")
-        return
-    #  When no more elements are there to put in data[]
-    if i >= n:
-        return
-    # current is included, put next at next location
-    data[index] = arr[i]
-    combination_util(arr, n, r, index + 1, data, i + 1)
-    # current is excluded, replace it with
-    # next (Note that i+1 is passed, but
-    # index is not changed)
-    combination_util(arr, n, r, index, data, i + 1)
-    # The main function that prints all combinations
-    # of size r in arr[] of size n. This function
-    # mainly uses combinationUtil()
-
-
-def print_combination(arr, n, r):
-    # A temporary array to store all combination one by one
-    data = [0] * r
-    # Print all combination using temporary array 'data[]'
-    combination_util(arr, n, r, 0, data, 0)
+# return all subset combinations of n element in given set of r element.
+
+
+def subset_combinations_dp(elements, n):
```

Comment:
No type hints, No docstring, no doctests, `n` is not a self-documenting variable name.

Why ask me to review this pull request when we have 200+ open pull requests that are more in alignment with CONTRIBUTING.md?

Reply:
Sorry for this mistake, but you previously said to do one thing at  a time.

Reply:
I did?

Reply:
yes